Source/Core/DolphinQt/SkylanderPortal/SkylanderPortalWindow.cpp

```diff
@@ -92,71 +228,367 @@ void SkylanderPortalWindow::CreateMainWindow()
+  m_game_filters[0]->setText(tr("Spyro's Adventure"));
+  m_game_filters[1]->setText(tr("Giants"));
+  m_game_filters[2]->setText(tr("Swap Force"));
+  m_game_filters[3]->setText(tr("Trap Team"));
+  m_game_filters[4]->setText(tr("Superchargers"));
+  search_checkbox_group->setLayout(search_checkbox_layout);
+  search_filters_layout->addWidget(search_checkbox_group);
+
+  auto* search_radio_group = new QGroupBox(tr("Element"));
+  auto* search_radio_layout = new QHBoxLayout();
+
+  auto* radio_layout_left = new QVBoxLayout();
+  for (int i = 0; i < 10; i += 2)
+  {
+    QRadioButton* radio = new QRadioButton(this);
+    radio->setProperty("id", i);
+    connect(radio, &QRadioButton::toggled, this, &SkylanderPortalWindow::RefreshList);
+    m_element_filter[i] = radio;
+    radio_layout_left->addWidget(radio);
+  }
+  search_radio_layout->addLayout(radio_layout_left);
+
+  auto* radio_layout_right = new QVBoxLayout();
+  for (int i = 1; i < 10; i += 2)
+  {
+    QRadioButton* radio = new QRadioButton(this);
+    radio->setProperty("id", i);
+    connect(radio, &QRadioButton::toggled, this, &SkylanderPortalWindow::RefreshList);
+    m_element_filter[i] = radio;
+    radio_layout_right->addWidget(radio);
+  }
+  search_radio_layout->addLayout(radio_layout_right);
+
+  m_element_filter[0]->setText(tr("All"));
+  m_element_filter[0]->setChecked(true);
+  m_element_filter[1]->setText(tr("Magic"));
+  m_element_filter[2]->setText(tr("Water"));
+  m_element_filter[3]->setText(tr("Tech"));
+  m_element_filter[4]->setText(tr("Fire"));
+  m_element_filter[5]->setText(tr("Earth"));
+  m_element_filter[6]->setText(tr("Life"));
+  m_element_filter[7]->setText(tr("Air"));
+  m_element_filter[8]->setText(tr("Undead"));
+  m_element_filter[9]->setText(tr("Other"));
+
+  search_radio_group->setLayout(search_radio_layout);
+  search_filters_layout->addWidget(search_radio_group);
+
+  auto* other_box = new QGroupBox(tr("Other"));
+  auto* other_layout = new QVBoxLayout;
+  connect(m_only_show_collection, &QCheckBox::toggled, this, &SkylanderPortalWindow::RefreshList);
+  other_layout->addWidget(m_only_show_collection);
+  other_box->setLayout(other_layout);
+  search_filters_layout->addWidget(other_box);
+
+  search_filters_layout->addStretch(50);
+
+  search_filters_group->setLayout(search_filters_layout);
+  search_layout->addWidget(search_filters_group);
+
+  search_layout->addWidget(m_skylander_list);
+
+  search_group->setLayout(search_layout);
+  main_layout->addWidget(search_group);
+
+  main_group->setLayout(main_layout);
+
+  return main_group;
+}
+
+void SkylanderPortalWindow::OnPathChanged()
+{
+  m_collection_path = m_path_edit->text();
+  RefreshList();
+}
+
+void SkylanderPortalWindow::SelectPath()
+{
+  QString dir = QDir::toNativeSeparators(DolphinFileDialog::getExistingDirectory(
+      this, tr("Select Skylander Collection"), m_collection_path));
+  if (!dir.isEmpty())
+  {
+    dir += QString::fromStdString("\\");
+    m_path_edit->setText(dir);
```

Comment:
This adds a \ char to the path, which makes the 'Only show files in collection' not work on unix machines